client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js

```diff
@@ -271,11 +298,99 @@ const getPackagesErrors = values =>
 		return errors;
 	} );
 
-export const getCustomsErrors = customs => {
-	if ( ! customs.dummyFieldChecked && customs.submitted ) {
-		return { dummyField: 'This must be selected to continue' };
-	}
-	return {};
+export const getCustomsErrors = (
+	packages,
+	customs,
+	destinationCountryCode,
+	destinationCountryName
+) => {
+	const usedProductIds = uniq(
+		flatten( map( packages, pckg => map( pckg.items, 'product_id' ) ) )
+	);
+
+	const valuesByProductId = zipObject( usedProductIds, fill( Array( usedProductIds.length ), 0 ) );
+	forEach( packages, pckg => {
+		forEach(
+			pckg.items,
+			( { quantity, product_id } ) =>
+				( valuesByProductId[ product_id ] += quantity * customs.items[ product_id ].value )
+		);
+	} );
+
+	const valuesByTariffNumber = {};
+	forEach( pick( customs.items, usedProductIds ), ( itemData, productId ) => {
+		if ( 6 === itemData.tariffNumber.length ) {
+			if ( ! valuesByTariffNumber[ itemData.tariffNumber ] ) {
+				valuesByTariffNumber[ itemData.tariffNumber ] = 0;
+			}
+			valuesByTariffNumber[ itemData.tariffNumber ] += valuesByProductId[ productId ];
+		}
+	} );
+
+	return {
+		packages: mapValues( packages, pckg => {
+			const errors = {};
+
+			if ( 'other' === pckg.contentsType && ! pckg.contentsExplanation ) {
+				errors.contentsExplanation = translate(
+					'Please describe what kind of goods this package contains'
+				);
+			}
+
+			if ( 'other' === pckg.restrictionType && ! pckg.restrictionExplanation ) {
+				errors.restrictionExplanation = translate(
+					'Please describe what kind of restrictions this package must have'
+				);
+			}
+
+			const classesAbove2500usd = new Set();
+			forEach( pckg.items, ( { product_id } ) => {
+				const { tariffNumber } = customs.items[ product_id ];
+				if ( 2500 < valuesByTariffNumber[ tariffNumber ] ) {
+					classesAbove2500usd.add( tariffNumber );
+				}
+			} );
+
+			if ( pckg.itn ) {
+				if ( ! /^AES X\d{14}$/.test( pckg.itn ) ) {
```

Comment:
In section 524.22 - 7 of the [ITN documentation](https://pe.usps.com/text/imm/immc5_010.htm), it says:

> [...] an alphanumeric Internal Transaction Number (ITN) as confirmation. The ITN consists of the letters “AES” followed by the ITN: for example, “AES X20080930987654.”

So it sounds like the ` X` is not part of every ITN number, and the last 15 characters can be alphanumeric.

Reply:
I've found this info: https://www.fedex.com/us/developer/WebHelp/ws/2015/html/WebServicesHelp/WSDVG/39_Electronic_Export_Information.htm

> The proper format for an ITN number is AES XYYYYMMDDNNNNNN where YYYYMMDD is date and NNNNNN are numbers generated by the AES.

Looks like the `X` is always part of the ITN, and we could even validate further since the first 8 digits are a date, but I think the current RegExp is reasonable to catch typos and strict enough.

Reply:
That's awesome 👍